fastchat/model/compression.py

```diff
@@ -117,15 +129,36 @@ def load_compress_model(model_path, device, torch_dtype, use_fast, revision="mai
             model_path,
             low_cpu_mem_usage=True,
             torch_dtype=torch_dtype,
-            trust_remote_code=True,
             revision=revision,
+            trust_remote_code=True,
         )
         # some models are loaded by AutoModel but not AutoModelForCausalLM,
         # such as chatglm, chatglm2
         try:
             model = AutoModelForCausalLM.from_config(config, trust_remote_code=True)
         except NameError:
             model = AutoModel.from_config(config, trust_remote_code=True)
+        if not max_gpu_memory:
+            max_memory = get_balanced_memory(
```

Comment:
I do not think this works with CPU only.

could you test this?
```
python3 -m fastchat.serve.cli --model-path lmsys/vicuna-7b-v1.3 --cpu --load-8bit
```

Reply:
New codes are tested on cpu, one gpu, and multi-gpu. All pass.